### ISSUE TITLE
upload_fileobj shouldn't convert all exceptions to ClientError

### ISSUE THREAD

**Issue by kneufeld** [terrycain/aioboto3#153]
* Async AWS SDK for Python version: 1.9.91
* Python version: 3.7
* Operating System: osx

### Description

I'm was trying to `cancel()` an upload task but the `CancelledError` got converted to a `ClientError` which means I can't safely do the right thing.

`inject.py` line `205` should just `raise` the previous exception, not convert it.

ie:

```
- raise ClientError({'Error': {'Code': '404', 'Message': 'Not Found'}}, 'HeadObject')
+ raise
```

**Comment by terrycain** [comment]
Correct, yeah i'll change that in a bit

**Comment by kneufeld** [comment]
Thanks a bunch, this package is extremely useful!

**Comment by terrycain** [comment]
> Thanks a bunch, this package is extremely useful!

Awesome :smile:, Should be fixed in v6.2.1
